plugins/modules/foreman_scap_content.py

```diff
@@ -0,0 +1,154 @@
+  - foreman.entity_state
+  - foreman.taxonomy
+'''
+
+EXAMPLES = '''
+- name: Create SCAP content
+  foreman_scap_content:
+    title: "Red Hat firefox default content"
+    scap_file: "/home/user/Downloads/ssg-firefox-ds.xml"
+    original_filename: "ssg-firefox-ds.xml"
+    organizations:
+      - "Default Organization"
+    locations:
+      - "Default Location"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    state: present
+
+- name: Update SCAP content
+  foreman_scap_content:
+    title: "Red Hat firefox default content"
+    updated_title: "Updated scap content title"
+    scap_file: "/home/user/Downloads/updated-ssg-firefox-ds.xml"
+    original_filename: "updated-ssg-firefox-ds.xml"
+    organizations:
+      - "Org One"
+      - "Org Two"
+    locations:
+      - "Loc One"
+      - "Loc Two"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    state: present
+
+- name: Delete SCAP content
+  foreman_scap_content:
+    title: "Red Hat firefox default content"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+import hashlib
+import os
+from ansible.module_utils.foreman_helper import ForemanTaxonomicEntityAnsibleModule
+
+
+class ForemanScapContentModule(ForemanTaxonomicEntityAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanScapContentModule(
+        argument_spec=dict(
+            updated_title=dict(type='str'),
+        ),
+        foreman_spec=dict(
+            title=dict(type='str', required=True),
+            original_filename=dict(type='str'),
+            scap_file=dict(type='path'),
+        ),
+        entity_key='title',
+        required_plugins=[('openscap', ['*'])],
+    )
+
+    with module.api_connection():
+        entity, module_params = module.resolve_entities()
+
+        if not module.desired_absent:
+            if not entity and 'scap_file' not in module_params:
+                module.fail_json(msg="Content of scap_file not provided. XML containing SCAP content is required.")
+
+            if 'scap_file' in module_params:
+                with open(module_params['scap_file']) as input_file:
+                    module_params['scap_file'] = input_file.read()
+
+            if entity and 'scap_file' in module_params:
+                digest = hashlib.sha256(module_params['scap_file'].encode("utf-8")).hexdigest()
+                # workaround for https://projects.theforeman.org/issues/29409
+                digest_stripped = hashlib.sha256(module_params['scap_file'].strip().encode("utf-8")).hexdigest()
+                if entity['digest'] in [digest, digest_stripped]:
+                    module_params.pop('scap_file')
+
+            if 'scap_file' in module_params and 'original_filename' not in module_params:
+                module_params['original_filename'] = os.path.basename(module_params['scap_file'])
```

Comment:
At this point, `module_params['scap_file']` does not contain the filename anymore.